lib/chef/provider/aws_internet_gateway.rb

```diff
@@ -0,0 +1,79 @@
+require 'chef/provisioning/aws_driver/aws_provider'
```

Comment:
No need to detach when updating with an already (different) attached vpc - from the docs:

> Attaches this internet gateway to the given VPC. If this gateway is already attached to a different VPC, it will be detached from that one first. If you pass nil, then this internet gateway will